src/ima.js

```diff
@@ -1041,6 +1041,8 @@ class Ima extends BasePlugin implements IMiddlewareProvider, IAdsControllerProvi
     this._contentComplete = true;
     if (this._currentAd && !this._currentAd.isLinear()) {
       this.reset();
+    } else if (!(this._adsManager && this._adsManager.getCuePoints().includes(-1))) {
+      this._stateMachine.goto(State.DONE);
```

Comment:
@yairans shouldn't it resolve here - https://github.com/kaltura/kaltura-player-js/blob/master/src/common/controllers/ads-controller.js#L325?
onPlaybackEnded should change the state to done

Reply:
@Yuvalke We check the ima's `done` state on bumper's `ADS_COMPLETED` (see https://github.com/kaltura/kaltura-player-js/blob/master/src/common/controllers/ads-controller.js#L295) which is before ima's `onPlaybackEnded` so it's too late to update the state then

Reply:
OK, so can you change the `this._adsManager && this._adsManager.getCuePoints().includes(-1)` to `hasPostroll` or something like that cause we use it multiple places